src/compiler/checker.ts

```diff
@@ -10415,6 +10415,11 @@ module ts {
                 });
             }
 
+            // Non-ambient classes cannot merge with interfaces.
+            if (!(node.flags & NodeFlags.Ambient) && getDeclarationOfKind(symbol, SyntaxKind.InterfaceDeclaration)) {

```

Comment:
Just check `symbol.flags & SymbolFlags.Interface` here instead of `getDeclarationOfKind`.